src/cpu/x64/jit_brgemm_inner_product_utils.cpp

```diff
@@ -336,8 +336,22 @@ int jit_brgemm_ip_conf_t::get_adjusted_oc_block() const {
     const bool is_f32_compute = !jbgp.is_bf32
             && everyone_is(f32, jbgp.src_dt, jbgp.wei_dt, jbgp.dst_dt);
     const bool is_avx512 = is_superset(jbgp.isa, avx512_core);
+    const bool is_avx2 = is_superset(jbgp.isa, avx2);
     const bool is_f32_compute_avx512 = is_f32_compute && is_avx512;
-
+    const bool is_f32_compute_avx2 = !is_avx512 && is_avx2 && is_f32_compute;
+    // Heuristic to unify oc block.
+    // batch size maybe affect oc_block, which would increased LLM peak USS memory.
+
+    // Use oc block to be 32 if weight size >= 8MB on amx bf16 to optimized memory consumption.
+    if (jbgp.is_amx && jbgp.wei_dt == bf16 && !jbgp.is_bf32
+        && jbgp.ic * jbgp.oc * 2 >= 8 * (1 << 20))
```

Comment:
Lets refactor a little bit for better understanding:
2 -> types::data_type_size(jbgp.wei_dt)
4 -> types::data_type_size(jbgp.wei_dt)